### ISSUE TITLE
dose it support for iOS now?

### ISSUE THREAD

**Issue by yunzhouhua** [wu9007/qrcode_scanner#51]

**Comment by rgbinnovation** [comment]
Not for me. Just tried it with await scanner.scan();, got the following error:

MissingPluginException (MissingPluginException(No implementation found for method scan on channel qr_scan))

**Comment by DroidGar** [comment]
> Not for me. Just tried it with await scanner.scan();, got the following error:
> 
> MissingPluginException (MissingPluginException(No implementation found for method scan on channel qr_scan))

same to, any help?

**Comment by Ravikumawat1990** [comment]
dose it support for iOS now?

**Comment by Dhua-v** [comment]
no It doesn't.

**Comment by Dhua-v** [comment]
> Why is this plugin marked as support IOS on pub.dev, when it does not support it?

Should we report it?

**Comment by wu9007** [comment]
iOS is implemented on `master` (qrscan 0.4.0).

Camera / album / path / bytes all go through AVFoundation + Vision. Add `NSCameraUsageDescription` and `NSPhotoLibraryUsageDescription` to `ios/Runner/Info.plist`. Minimum iOS 12.

This is still a one-shot `scan()` (open camera → first decode → return `String?` → close), not an embedded preview widget. If you need a scan window or continuous scan, use [mobile_scanner](https://pub.dev/packages/mobile_scanner).

Install from git until pub.dev ships 0.4.0 (pub still has 0.3.3):

```yaml
dependencies:
  qrscan:
    git:
      url: https://github.com/wu9007/qrcode_scanner.git
      ref: master
```

Closing as completed.

**Comment by wu9007** [comment]
@yunzhouhua yes, iOS is implemented on `master` (0.4.0): AVFoundation + Vision, channel `qr_scan`. The years of “does it support iOS?” were correct — pub 0.2.x / 0.3.3 did not.

Install from git until pub.dev has 0.4.0. Closing as completed.
